### DISCUSSION TITLE
Enable partial matching of keys for setQueryData

### DISCUSSION THREAD

**Discussion by villesau** [TanStack/query#3239] [Ideas] [answered]
Hi,

based on my limited experience, most of the time when dealing with lists, partial matching is the desired behaviour. We pull data based on order, but when updating it, the component doing that should not care about the order. Only about the type. So while we have cache key such as `["items", "most_recent"]`, `setQueryData` should be possible with `["items"]` only. This is repeating use case in the lists I'm dealing with and wiring up the order information would mean a lot of variables passed around since otherwise ordering is irrelevant from the individual items perspective.

If you want to be explicit about it, some type of wild card functionality could work, although I think it would be pretty sensible default behaviour too: 

`["items", "*"]`

or:

```
import { any } from 'react-query'
["items", any]
```

**Comment by TkDodo** [comment, marked as answer]
[setQueriesData](https://react-query.tanstack.com/reference/QueryClient#queryclientsetqueriesdata) is what you are looking for.

**Reply by villesau** [reply]
Oh, that's exactly true, thanks! Documentation wise would be probably good to drop keywords like "partial" in the description of `setQueriesData`. Maybe also cross-mention next to some `setQueryData` examples? Also cross links in some discussions could make sense, there's some outdated information. Couldn't spot this in any examples I checked.

**Reply by villesau** [reply]
@TkDodo I proposed minor changes to documentation to make this a bit clearer and easier to find: https://github.com/tannerlinsley/react-query/pull/3244
